spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/web/reactive/ReactiveWebServerFactoryAutoConfigurationTests.java

```diff
@@ -97,6 +102,36 @@ public void defaultWebServerIsTomcat() {
 								.isInstanceOf(TomcatReactiveWebServerFactory.class));
 	}
 
+	@Test
+	public void tomcatConnectorCustomizerBeanIsAddedToFactory() {
+		WebApplicationContextRunner runner = new WebApplicationContextRunner(
+				AnnotationConfigServletWebServerApplicationContext::new)
+						.withConfiguration(AutoConfigurations
+								.of(ReactiveWebServerFactoryAutoConfiguration.class))
+						.withUserConfiguration(
+								TomcatConnectorCustomizerConfiguration.class);
+		runner.run((context) -> {
+			TomcatServletWebServerFactory factory = context
+					.getBean(TomcatServletWebServerFactory.class);
+			assertThat(factory.getTomcatConnectorCustomizers()).hasSize(1);
+		});
+	}
+
+	@Test
+	public void tomcatContextBeanIsAddedToFactory() {
+		WebApplicationContextRunner runner = new WebApplicationContextRunner(
+				AnnotationConfigServletWebServerApplicationContext::new)
+						.withConfiguration(AutoConfigurations
+								.of(ReactiveWebServerFactoryAutoConfiguration.class))
+						.withUserConfiguration(
+								TomcatContextCustomizerConfiguration.class);
+		runner.run((context) -> {
+			TomcatServletWebServerFactory factory = context
+					.getBean(TomcatServletWebServerFactory.class);
+			assertThat(factory.getTomcatContextCustomizers()).hasSize(1);
```

Comment:
`assertThat(factory.getTomcatContextCustomizers()).hasSize(1);` is working fine in `TomcatServletWebServerFactory`  case. If i remove `.withUserConfiguration(TomcatContextCustomizerConfiguration.class)` from test assertion going to fail.  If we doesn't define `TomcatContextCustomizer ` into `TomcatContextCustomizerConfiguration`  assertion is also failed because tomcatContextCustomizers list is zero.